server/src/main/java/org/elasticsearch/health/HealthIndicatorDetails.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public interface HealthIndicatorDetails extends ToXContentFragment {
```

Comment:
I wonder if we could make this a pojo with a details Map<String, Object> inside or this might not be enough in certain implementations?

Reply:
I think there are likely to be more complex details, where the map-form will hinder code readability. But I think we could add a `SimpleHealthIndicatorDetails` for the simple cases.